src/passes/OptimizeInstructions.cpp

```diff
@@ -3597,6 +3597,57 @@ struct OptimizeInstructions
         }
       }
     }
+    {
```

Comment:
The title of this PR makes sense for the NaN opts, but the copysign stuff seems unrelated. It could be in a separate PR?

Reply:
Yes, you can do a separate PR. I decided in one, so they are connected. In fact, copysign is a more general, but separate case. I can put copysign rules in a separate PR

Reply:
Maybe I just don't see the connection. But the code is separate, tests separate, etc., so it seems like separate PRs are best.

Reply:
Sure! Will do

Reply:
done!